D2L.Hypermedia.Siren/JsonUtilities.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+
+namespace D2L.Hypermedia.Siren {
+
+	public static class JsonUtilities {
+
+		public static void WriteJsonArray( JsonWriter writer, string propertyName, string[] values ) {
+			if( values == null || values.Length == 0 ) {
+				return;
+			}
+
+			writer.WritePropertyName( propertyName );
+			writer.WriteStartArray();
+
+			foreach( string value in values ) {
+				writer.WriteValue( value );
+			}
+
+			writer.WriteEndArray();
+		}
+
+		public static void WriteJsonSerializable( JsonWriter writer, string propertyName, IEnumerable<ISirenSerializable> values ) {
+			if( values == null || !values.Any() ) {
+				return;
+			}
+
+			writer.WritePropertyName( propertyName );
+			writer.WriteStartArray();
+
+			foreach( ISirenSerializable serializable in values ) {
+				writer.WriteRawValue( serializable.ToJson() );

```

Comment:
Would be neato if there were an `ToJson( JsonTextWriter writer )` perhaps

Reply:
It's a little odd since it's `void ToJson( JsonTextWriter )` rather than returning a string, but it still makes sense I think. Coming up!